src/framework/execution/__tests__/validation.test.ts

```diff
@@ -0,0 +1,78 @@
+import { IntegrationStep, IntegrationStepStartStates } from '../types';
+
+import { validateStepStartStates } from '../validation';
+
+describe('validateStepStartStates', () => {
+  test('throws error if step is provided in start states', () => {
+    const states: IntegrationStepStartStates = {
+      a: {
+        disabled: false,
+      },
+      b: {
+        disabled: true,
+      },
+    };
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        executionHandler: jest.fn(),
+      },
+    ];
+
+    expect(() => validateStepStartStates(steps, states)).toThrow(
+      `Invalid step id "b" found in start states.`,
+    );
+  });
+
+  test('throws steps are not accounted for in start states', () => {
+    const states: IntegrationStepStartStates = {
+      a: {
+        disabled: false,
+      },
+    };
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'b',
+        name: 'b',
+        types: [],
+        executionHandler: jest.fn(),
+      },
+      {
+        id: 'c',
+        name: 'c',
+        types: [],
+        executionHandler: jest.fn(),
+      },
+    ];
+
+    expect(() => validateStepStartStates(steps, states)).toThrow(
+      `Steps not defined in start states found: "b", "c"`,
```

Comment:
```suggestion
      `Start states not found for : "b", "c"`,
```
I would prefer this because I think of the steps as being the authoritative set of things to do, with the start states as simply being input to those steps.